### ISSUE TITLE
Transforms influencing priors

### ISSUE THREAD

**Issue by frgsimpson** [GPflow/GPflow#1176]
It seems that adding a Uniform prior to a Parameter with a transform leads to a non-uniform prior.  This arises because the prior probability density is evaluated on the unconstrained parameter, via the log_det_jacobian, where the prior is no longer uniform.

Example code provided below:

```python
from gpflow import Parameter
from gpflow.kernels import Matern12

import numpy as np
from gpflow.utilities import positive
from tensorflow_probability.python.distributions import Uniform

LENGTHSCALES = [0.1, 1.0]
wide_uniform_prior = Uniform(low=np.float64(1e-10), high=np.float64(1e10))

kernel = Matern12()
kernel.lengthscale = Parameter(1.0, transform=positive(), prior=wide_uniform_prior)

for length in LENGTHSCALES:
    kernel.lengthscale.assign(length)
    print(kernel.lengthscale.log_prior().numpy(), 'at l=', length)
```

-----------------------------------------------------------------
Output:
 (Tested on 'develop':  ef5a7ca1aad46878735a6e30df3c43480cb7dd6b)
-----------------------------------------------------------------
-25.37801937681603 at l= 0.1
-23.48452607532754 at l= 1.0

**Comment by frgsimpson** [comment]
Related to https://github.com/GPflow/GPflow/pull/1177 which makes explicit how the prior is defined and evaluated.

**Comment by st--** [comment]
@frgsimpson has this issue been resolved by your PR #1177 ?
